common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -625,6 +626,72 @@ def test_reinstate_entitlement(self):
         )
         assert results == CourseEntitlementSerializer(reinstated_entitlement).data
 
+    def test_reinstate_refundable_entitlement(self):
+        """ Verify that an entitlement that is refundable stays refundable when support reinstates it. """
+        enrollment = CourseEnrollmentFactory(user=self.user, is_active=True, course=CourseOverviewFactory(start=now()))
+        fulfilled_entitlement = CourseEntitlementFactory.create(
+            user=self.user, enrollment_course_run=enrollment
+        )
+        assert fulfilled_entitlement.is_entitlement_refundable() is True
+        url = reverse(self.ENTITLEMENTS_DETAILS_PATH, args=[str(fulfilled_entitlement.uuid)])
+
+        update_data = {
+            'expired_at': None,
+            'enrollment_course_run': None,
+            'support_details': [
+                {
+                    'unenrolled_run': str(enrollment.course.id),
+                    'action': CourseEntitlementSupportDetail.REISSUE,
+                    'comments': 'Severe illness.'
+                }
+            ]
+        }
+
+        response = self.client.patch(
+            url,
+            data=json.dumps(update_data),
+            content_type='application/json'
+        )
+        assert response.status_code == 200
+
+        reinstated_entitlement = CourseEntitlement.objects.get(
+            uuid=fulfilled_entitlement.uuid
+        )
+        assert reinstated_entitlement.refund_locked is False
```

Comment:
Update the logic here to also confirm that the entitlement is still refundable

`assert fulfilled_entitlement.is_entitlement_refundable() is True`